src/wallet/wallettool.cpp

```diff
@@ -140,11 +140,6 @@ bool ExecuteWalletToolFunc(const std::string& command, const std::string& name)
             tfm::format(std::cerr, "Error: no wallet file at %s\n", name);
             return false;
         }
-        bilingual_str error;
-        if (!WalletBatch::VerifyEnvironment(path, error)) {
-            tfm::format(std::cerr, "%s\nError loading %s. Is wallet being used by other process?\n", error.original, name);
-            return false;
-        }
```

Comment:
How is it unrelated? This function is being removed from `WalletBatch`.

Reply:
I was wrong. VerifyEnvironment is called via `LoadWallet -> LoadWalletInternal -> CWallet::Verify`, which is different from the `LoadWallet` call below.

So why is it ok to remove this call to `VerifyEnvironment`?

Reply:
`VerifyEnvironment` would just call `BerkeleyEnvironment::Open`. A couple of other things also call that in their normal operation. In particular, `CWallet::LoadWallet` creates a `WalletBatch` which creates a `BerkeleyBatch` which calls `BerkeleyEnvironment::Open` in its constructor.

In the cases where we need to open the environment and aren't using the higher level stuff, a direct call to `BerkeleyEnvironment::Open` is needed which is why it is added to `SalvageWallet`.

Reply:
With "unrelated" I meant "can be split up into its own atomic and logical commit without depending on other changes in this pull request".

See the following commit on current master, which compiles fine and runs all tests fine as well:

```
$ git show 
commit 43d120ec8a7a999a48ba31e2621efec06ca668f6 (HEAD)
Author: MarcoFalke <falke.marco@gmail.com>
Date:   Thu Jul 2 07:53:05 2020 -0400

    wallet: Remove redundant WalletBatch::VerifyEnvironment
    
    It is redundant because LoadWallet will call CWallet::LoadWallet ->
    LoadWalletInternal -> CWallet::Verify, which calls VerifyEnvironment
    
    For SalvageWallet it is redundant with the newly added call to Open

diff --git a/src/wallet/salvage.cpp b/src/wallet/salvage.cpp
index d42950ee42..e6e62332c0 100644
--- a/src/wallet/salvage.cpp
+++ b/src/wallet/salvage.cpp
@@ -20,6 +20,11 @@ bool RecoverDatabaseFile(const fs::path& file_path)
     std::string filename;
     std::shared_ptr<BerkeleyEnvironment> env = GetWalletEnv(file_path, filename);
 
+    if (!env->Open(true /* retry */)) {
+        tfm::format(std::cerr, "Error initializing wallet database environment %s!", env->Directory());
+        return false;
+    }
+
     // Recovery procedure:
     // move wallet file to walletfilename.timestamp.bak
     // Call Salvage with fAggressive=true to
diff --git a/src/wallet/wallettool.cpp b/src/wallet/wallettool.cpp
index 77ed6beb5d..430a866a7c 100644
--- a/src/wallet/wallettool.cpp
+++ b/src/wallet/wallettool.cpp
@@ -140,11 +140,6 @@ bool ExecuteWalletToolFunc(const std::string& command, const std::string& name)
             tfm::format(std::cerr, "Error: no wallet file at %s\n", name);
             return false;
         }
-        bilingual_str error;
-        if (!WalletBatch::VerifyEnvironment(path, error)) {
-            tfm::format(std::cerr, "%s\nError loading %s. Is wallet being used by other process?\n", error.original, name);
-            return false;
-        }
 
         if (command == "info") {
             std::shared_ptr<CWallet> wallet_instance = LoadWallet(name, path);
diff --git a/test/functional/tool_wallet.py b/test/functional/tool_wallet.py
index 524e1593ba..18f0beb598 100755
--- a/test/functional/tool_wallet.py
+++ b/test/functional/tool_wallet.py
@@ -71,8 +71,7 @@ class ToolWalletTest(BitcoinTestFramework):
         self.assert_raises_tool_error('Error: two methods provided (info and create). Only one method should be provided.', 'info', 'create')
         self.assert_raises_tool_error('Error parsing command line arguments: Invalid parameter -foo', '-foo')
         self.assert_raises_tool_error(
-            'Error initializing wallet database environment "{}"!\nError loading wallet.dat. Is wallet being used by other process?'
-            .format(os.path.join(self.nodes[0].datadir, self.chain, 'wallets')),
+            'Error loading wallet.dat. Is wallet being used by another process?',
             '-wallet=wallet.dat',
             'info',
         )

Reply:
Maybe if I need to push again.